### ISSUE TITLE
[Feature] Negotiated rate on port / advertising + partner advertising

### ISSUE THREAD

**Issue by smathev** [tomaae/homeassistant-mikrotik_router#146]
Hi.

So, recently installed this to get more information about my LAN-network. What I was really hoping for was the ability to monitor the negotiated rate between a switch and a port.

However, that doesn't seem to be info that is currently being pulled - is there any way to get that information?

It would seem it belong in the 'mikrotik_port'-entity values, but it doesn't seem to be there.

Thanks in advance for any answer you can provide.

**Comment by tomaae** [comment]
Where is that info available in routeros cli? I cant find it.

**Comment by smathev** [comment]
Hi.

Thanks for getting back to me!

Haven't checked in Winbox, but this seems to be a discussion of how to get the information in RouterOs terminal.

Does that help?

**Comment by tomaae** [comment]
If you tried to attach something in you last comment, I cannot see anything.
Yes, it has to be accessible using terminal. Winbox is not really relevant to how integration functions.

**Comment by smathev** [comment]
Yeah, sorry.

It was this link, I meant to attach: https://forum.mikrotik.com/viewtopic.php?t=58031

I'm unsure what you mean, when refering to terminal - there is a terminal within the WinBox interface and I'm assuming that's sufficient?

Anyway, I hope it's possible - and, if not, I still think your development and project is very useful for me :)

**Comment by tomaae** [comment]
looks doable

**Comment by tomaae** [comment]
can you test master?

**Comment by tomaae** [comment]
looks like I wont be able to get rate on bonds, which is unfortunate.

**Comment by smathev** [comment]
Hi Tomaae.

I would love to test, but I'm uncertain as to what you're asking me to do in 'testing master'. I'm using HACS to maintain my installation of your integration.

Could you point me to how I should update it, to 'test master'?

**Comment by tomaae** [comment]
you can choose master in version selection

**Comment by smathev** [comment]
Perfect ... I mean, I haven't quite tested it thoroughly, but when set up it reported the correct rate that was negotiated. 
When I changed the rate of one of the interfaces and reloaded the entity-state to check if it updated is was already updated.

'looks like I wont be able to get rate on bonds, which is unfortunate.'  - I'm not sure what that is, since I don't think I use it (or at least need the report of the rate in this issue). In short, that isn't an issue for me in relation to what I asked in this feature request :)

In short, I think this is doing exactly what I asked!
You can close this issue, I will open a new issue with reference to this one if there are any unexpected issues. 

Thank you for the quick turn around, perfect!
